lib/improver/tests/utilities/test_cube_manipulation.py

```diff
@@ -527,5 +527,63 @@ def test_cubelist_different_var_names(self):
         self.assertIsInstance(result, Cube)
 
 
+class Test_add_renamed_cell_method(IrisTest):
+    """Class to test the add_renamed_cell_method function"""
+
+    def setUp(self):
+        '''Set up input cube for tests'''
+        self.cube = set_up_temperature_cube()
+        self.cell_method = iris.coords.CellMethod(method='mean', coords='time')
+        self.cube.add_cell_method(self.cell_method)
+
+    def test_basic(self):
+        """Basic test for one cell method on input cube"""
+        add_renamed_cell_method(self.cube, self.cell_method, 'weighted_mean')
+        expected_cell_method = iris.coords.CellMethod(method='weighted_mean',
+                                                      coords='time')
+        self.assertEqual(self.cube.cell_methods, (expected_cell_method,))
+
+    def test_only_difference_is_name(self):
+        """Testing that the input cell method and the new cell method only
+           differ by name"""
+        add_renamed_cell_method(self.cube, self.cell_method, 'weighted_mean')
+        expected_cell_method = iris.coords.CellMethod(method='weighted_mean',
+                                                      coords='time')
+        self.assertEqual(self.cube.cell_methods, (expected_cell_method,))
+        new_cell_method = self.cube.cell_methods[0]
+        self.assertEqual(self.cell_method.coord_names,
+                         new_cell_method.coord_names)
+        self.assertEqual(self.cell_method.intervals, new_cell_method.intervals)
+        self.assertEqual(self.cell_method.comments, new_cell_method.comments)
+
+    def test_no_cell_method_in_input_cube(self):
+        """Testing that when there are no cell methods on the input cube then
+           the new cell method still gets added as expected."""
```

Comment:
I also think the margin is distant (as do many of my fellow follow on lines).

Reply:
Moved in